File: touch_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def touch_file(file_path, config):
    if not check_file_permission(file_path, "write", config):
        raise PermissionError(f"Write access denied for {file_path}")
    file_path = Path(file_path).resolve()
    file_path.touch()
    return f"Touched {file_path}"

File: ts-tools/touch_tool.ts

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function touchFile(filePath: string, config: any): Promise<string> {
    return new Promise((resolve, reject) => {
        checkFilePermission(filePath, "write", config)
            .then(() => {
                const resolvedPath = path.resolve(filePath);
                const time = new Date();
                fs.utimes(resolvedPath, time, time, (err) => {
                    if (err) {
                        fs.writeFile(resolvedPath, '', (err) => {
                            if (err) {
                                reject(err);
                            } else {
                                resolve(`Touched ${filePath}`);
                            }
                        });
                    } else {
                        resolve(`Touched ${filePath}`);
                    }
                });
            })
            .catch(reject);
    });
}
